### DISCUSSION TITLE
Use faker.Js to generate responses

### DISCUSSION THREAD

**Discussion by StanleyMasinde** [opensource254/mock-api#29] [Ideas]
What do you guys think of using faker.js to generate responses as opposed to hard coded values.
We need to ensures consistency e.g

1. A user makes a request to get a list of users and gets a repsonse as below
```json
[
  {
      "id": "1",
      "name": "John Doe"
  },
  {
      "id": "2",
      "name": "Ashley Doe"
  }
]
```
2. If the user makes a request for user with id 1, they should get John Doe

CC @opensource254/backend @miami78 @cyrus43

**Comment by larrymiami** [comment]
@StanleyMasinde i think this is a good idea but quick question, how do we ensure the consistency of data? doesn't generating the data using faker change every time one requests for it?

**Reply by StanleyMasinde** [reply]
That's the thing I need us to brainstorm.
We could use sessions for that maybe?

**Reply by Donstesh** [reply]
I think since it is giving random fake values each time you post a request it will come with different values.

**Reply by larrymiami** [reply]
Doesnt sessions have limited lifetimes?

**Reply by StanleyMasinde** [reply]
That's right @Donstesh Here is what I'm proposing
When the user makes a request for the list of users, The application will create a session with the generated values.
When the same user tries to get the user with id 1, we get that from the sesssion.

What do you think of that?

**Reply by larrymiami** [reply]
sounds valid, we can use redis for that

**Reply by Donstesh** [reply]
So @StanleyMasinde you are proposing storing the sessions data for first request if successful. Is that achievable? Just thinking 🤔

**Reply by StanleyMasinde** [reply]
@Donstesh we can try and see.

**Comment by cyrus43** [comment]
If a user makes a request to get a list of users, the API should return a
list of user objects containing the expected information, such as the
user's id and name. If the user then makes a request for a specific user
with a certain id, the API should return the corresponding user object with
the same id and name as in the previous list.

My take, It would be appropriate to hard-code the values for the user ids
and names in the API, rather than using a library like faker.js to generate
randomized values. This will ensure that the same user object is returned
for a given id, and that id and name match across different requests.

**Reply by Donstesh** [reply]
@cyrus43 What if you are going to test more than 2000 values will you hard code all that 😩🥱

**Reply by cyrus43** [reply]
we can use redis as @larrymiami suggested

**Reply by StanleyMasinde** [reply]
I think so too. We could try redis and see how it goes.
